### PR TITLE
TANZU-3572: handle Spring Cloud Gateway failovers

### PR DESCRIPTION
**Pull request by neowulf**

### CODE REVIEW

File: where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ metadata:
   name: where-for-dinner-scg-route
   annotations:
    healthcheck.gslb.tanzu.vmware.com/service: where-for-dinner-ui
```
Comment (aaronjwood) [inline where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml:6]: Is it intentional that the health check goes to the UI service?

Comment (Joey-Rose) [inline where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml:6]: this should be spring-cloud-gateway

Comment (neowulf) [reply where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml]: good catch!

Comment (neowulf) [reply where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml]: good catch!

**Comment by aaronjwood** [comment]
lgtm, thx!

**Comment by neowulf** [comment]
from @gm2552 

> What are you trying to monitor here? Are you trying to monitor that the gateway is working, or that the UI app is working? If you are trying to monitor the gateway, there is an actuator that should be acceible.
> I stand corrected on the Gateway actuator as I'm not sure you will be able to access it using the `spring-cloud-gateway` K8s service name because the actuator is on another port that is not accessible via the K8s service. Based on our Zoom conversation yesterday, the changes here will work for your use case, however `/diningsearch/health` is technically not a URL that is part of the UI app (it works more on accident as opposed to on purpose). I would suggest making the path `/` instead of `diningsearch/health`.

**Comment by neowulf** [comment]
updated the healthcheck path from `/diningsearch/health` to `/`

**Review by gm2552 — APPROVED** [review]

File: where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml [hunk 2]
```diff
@@ -3,9 +3,9 @@ kind: HTTPRoute
 metadata:
   name: where-for-dinner-scg-route
   annotations:
-   healthcheck.gslb.tanzu.vmware.com/service: where-for-dinner-ui
-   healthcheck.gslb.tanzu.vmware.com/path: /health
-   healthcheck.gslb.tanzu.vmware.com/port: "8080"
+   healthcheck.gslb.tanzu.vmware.com/service: spring-cloud-gateway
```
[the remark this answers is not in the document]
Comment (gm2552) [reply where-for-dinner/spaces-beta3-deployment/aws/routes/k8sGatewayRoutes.yaml:6]: Looks good now.

**Pull request merged**